File: pagamento-oneclick/su-sito-merchant/pagamenti-successivi/tramite-redirezione/php/pagamento-successivo.php
                                            
<?php

// Pagamento OneClik - Pagamenti successivi - Tramite redirezione - Avvio pagamento

$requestUrl = "https://int-ecommerce.nexi.it/ecomm/ecomm/DispatcherServlet";
$merchantServerUrl = "https://" . $_SERVER['HTTP_HOST'] . "/xpay/php/pagamento_semplice/one_click/";

//PARAMETRI PER CALCOLO MAC
$codTrans = "TESTPS_" . date('YmdHis');
$importo = 5000; /* <-- 5000 = 50,00 EURO (prime due cifre a destra per i centesimi) */
$chiaveSegreta = "<CHIAVE SEGRETA PER CALCOLO MAC>";
$divisa = "EUR"; /* <-- EUR oppure 978 */
$numContratto = "NC_TEST_" . date('YmdHis');

//CALCOLO MAC
$mac = sha1('codTrans=' . $codTrans . 'divisa=' . $divisa . 'importo=' . $importo . $chiaveSegreta);

//Param Obbligatori
$requestParams = array(
    'importo' => $importo,
    'alias' => "<ALIAS>",
    'divisa' => $divisa,
    'codTrans' => $codTrans,
    'mac' => $mac,
    'url' => "<URL DI RITORNO DOPO AVER COMPLETATO IL PAGAMENTO>", //necessita HTTP:// oppure HTTPS://
    'url_back' => "<URL DI RITORNO (pagina iniziale)", //necessita HTTP:// oppure HTTPS://
    'urlpost' => "<URL NOTIFICA POST ESITO TRANSAZIONE>", //necessita HTTP:// oppure HTTPS://
    'num_contratto' => $numContratto,
    'tipo_servizio' => 'paga_oc3d',
    'tipo_richiesta' => 'PR', /* <-- PR = Pagamento Ricorrente */
);

?>

<html>
    <head></head>
    <body>
        <form method='POST' action='<?php echo $requestUrl ?>'>
            <?php foreach ($requestParams as $name => $value) { ?>
                <input type='hidden' name='<?php echo $name; ?>' value='<?php echo htmlentities($value); ?>' />
            <?php } ?>
            
            <input type='submit' value='VAI ALLA PAGINA DI CASSA' />
        </form>
    </body>
</html>
                    
                

File: pagamento-oneclick/su-sito-merchant/pagamenti-successivi/tramite-redirezione/java/pagamento-successivo.java
                                            
// Pagamento OneClik - Pagamenti successivi - Tramite redirezione - Avvio pagamento

import java.util.Map;
import java.util.HashMap;
import java.security.MessageDigest;
import java.util.Date;
import java.text.SimpleDateFormat;
import java.net.URLEncoder;

public class pagamentiSuccessivi-TramireRedirezione {

  public static void main(String[] args) throws Exception {

    String ALIAS = "<ALIAS>"; // Sostituire con il valore fornito da Nexi
    String CHIAVESEGRETA = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi
    String GRUPPO = "<GRUPPO>"; // Sostituire con il valore fornito da Nexi

    // questi sono i parametri in ingresso della richiesta
    Map < String, String > paramFromRequest = new HashMap < String, String > ();

    String HTTP_HOST = "my-server.example.tdl";

    String requestUrl = "https://int-ecommerce.nexi.it/ecomm/ecomm/DispatcherServlet";
    String merchantServerUrl = "https://" + HTTP_HOST + "/xpay/pagamento_semplice_java/one_click/";

    // PARAMETRI PER CALCOLO MAC
    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
    Date date = new Date();

    String codTrans = "TESTPS_" + dateFormat.format(date);
    String divisa = "EUR";
    String importo = "5000";

    // CALCOLO MAC
    String stringaMac = "codTrans=" + codTrans +
            "divisa=" + divisa +
            "importo=" + importo +
            CHIAVESEGRETA;

    String macCalculated = hashMac(stringaMac);

    String numContratto;
    String tipoRichiesta;
    if (paramFromRequest.get("numContratto")!=null) {
        numContratto = paramFromRequest.get("numContratto");
        tipoRichiesta = "PR";
    } else {
        numContratto = "NC_TEST_" + dateFormat.format(date);
        tipoRichiesta = "PP";
    }

    // Param Obbligatori
    HashMap<String, String> requestParams = new HashMap<String, String>();

    requestParams["alias"] = ALIAS;
    requestParams["importo"] = importo;
    requestParams["divisa"] = divisa;
    requestParams["codTrans"] = codTrans;
    requestParams["url"] = merchantServerUrl + "esito.html";
    requestParams["url_back"] = merchantServerUrl + "annullo.html";
    requestParams["mac"] = macCalculated;
    requestParams["urlpost"] = merchantServerUrl + "notifica.html";
    requestParams["num_contratto"] = numContratto;
    requestParams["tipo_servizio"] = "paga_oc3d";
    requestParams["tipo_richiesta"] = tipoRichiesta;    
            
    /**
     * Creare un form html con metodo post verso requestUrl con campi hidden contenenti requestParams
     */
  }

  public static String hashMac(String stringaMac) throws Exception {
      MessageDigest digest = MessageDigest.getInstance("SHA-1");
      byte[] in = digest.digest(stringaMac.getBytes("UTF-8"));

      final StringBuilder builder = new StringBuilder();
      for(byte b : in) {
          builder.append(String.format("%02x", b));
      }
      return builder.toString();
  }

}
                    
                